tcms/bugs/tests/test_views.py

```diff
@@ -247,3 +247,23 @@ def test_add_reopen_bug_comment(self):
 
         self.assertEqual(comments.count(), old_comment_count + 1)
         self.assertEqual(comments.last().comment, _('*bug reopened*'))
+
+
+class TestSearch(LoggedInTestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        user_should_have_perm(cls.tester, 'bugs.view_bug')
+
+    def test_initial_form_field_states(self):
+        ProductFactory()
+        VersionFactory()
+        BuildFactory()
+
+        response = self.client.get(reverse('bugs-search'))
+        fields = response.context['form'].fields
+
+        self.assertEqual(fields['version'].queryset.count(), 0)
+        self.assertEqual(fields['build'].queryset.count(), 0)
+        self.assertNotEqual(fields['product'].queryset.count(), 0)
```

Comment:
This doesn't look right. The product field should be displaying all available products while version and build are dynamically changed via JavaScript so they are empty initially.

Reply:
I think that's what my code is actually checking for.

build_count == 0
version_count == 0
product_count != 0

I can't hardcode the exact count, as it seems extra products were created in the test database (product queryset count actually returned 3 in my tests instead of 1).

Reply:
I've changed it to be clearer in the new push.

Reply:
> I've changed it to be clearer in the new push.

Thanks. I didn't see the `NotEqual` part before. Now is more clear.